lib/ansible/template/__init__.py

```diff
@@ -595,7 +595,7 @@ def _lookup(self, name, *args, **kwargs):
                                        "original message: %s" % (name, type(e), e))
                 ran = None
 
-            if ran and not allow_unsafe:
+            if ran:
```

Comment:
I'm unclear how this fixes things? By removing this, the `ran` result is always going to be wrapped in `UnsafeProxy()`.  It then relies on setting the flag on the context, but the value is still considered unsafe even if the environment isn't.

Reply:
It's exactly as you say. After further research my issue is caused by `wantlist` not being honored when `allow_unsafe` is set, which caused the `from_yaml` filter to return the list as is.

Reply:
I.e. `lookup()` returns a list when `allow_unsafe=True`, which doesn't match the behavior when `allow_unsafe=False`. Which was the previous behavior.